pkg/apis/machine/validation/machinedeployment.go

```diff
@@ -51,6 +60,23 @@ func validateUpdateStrategy(spec *machine.MachineDeploymentSpec, fldPath *field.
 			}
 		}
 	}
+
+	if spec.Strategy.Type == machine.InPlaceUpdateMachineDeploymentStrategyType {
+		if spec.Strategy.InPlaceUpdate == nil {
+			allErrs = append(allErrs, field.Required(fldPath.Child("strategy.inPlaceUpdate"), "InPlaceUpdate parameter cannot be nil for in-place update strategy"))
+		} else {
+			if !canConvertIntOrStringToInt32(spec.Strategy.InPlaceUpdate.MaxUnavailable, int(spec.Replicas)) {
+				allErrs = append(allErrs, field.Required(fldPath.Child("strategy.inPlaceUpdate.maxUnavailable"), "unable to convert maxUnavailable to int32"))
+			}
+			if !canConvertIntOrStringToInt32(spec.Strategy.InPlaceUpdate.MaxSurge, int(spec.Replicas)) {
+				allErrs = append(allErrs, field.Required(fldPath.Child("strategy.inPlaceUpdate.maxSurge"), "unable to convert maxSurge to int32"))
+			}
+			if spec.Strategy.InPlaceUpdate.OrchestrationType != machine.OrchestrationTypeAuto && spec.Strategy.InPlaceUpdate.OrchestrationType != machine.OrchestrationTypeManual {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("strategy.inPlaceUpdate.orchestrationType"), spec.Strategy.InPlaceUpdate.OrchestrationType, "orchestrationType must be either Auto or Manual"))
+			}
```

Comment:
This block is used with `InPlaceUpdate` as well as `RollingUpdate`
Can you please add this in a common block?

Reply:
Addressed all the comments in 52fd0520762771b7a91b522a35b302a65775a906. PTAL